### ISSUE TITLE
Error: the surface type is not appropriate for the operation

### ISSUE THREAD

**Issue by acrodrig** [fuzhenn/node-svg2img#20]
Getting the following error:

```
Error: the surface type is not appropriate for the operation
    at PNGStream._read 
```

When I try to execute this program. Any ideas what could be the problem?

```
svg2img(__dirname + "/image.svg", function(error, buffer) {
    if (error) console.log(error);
    fs.writeFileSync("/tmp/image.png", { width: 100, height:100 }, buffer);
});
```

**Comment by fuzhenn** [comment]
a related issue from upsteam: https://github.com/Automattic/node-canvas/issues/344
may bring some clue

**Comment by almarcTheSun** [comment]
Did you manage to solve that?

**Comment by jamesmorgan** [comment]
I get this as well - no sign of a fix?

**Comment by fuzhenn** [comment]
According to the https://github.com/Automattic/node-canvas/issues/344, this is caused by float (not integer) canvas width and height.

Could someone upload a reproduction svg? I'll try to fix this error, thanks.

**Comment by RGunning** [comment]
seems that instantiating the canvas with a width and height fixes the issue. Made a quick PR that solves it for me.

here is an svg that can be used for testing which broke before the fix
`<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M19.247 6.24l-3.836 3.851 1.86 8.07-1.335 1.339-3.28-6.645-2.596 2.605v2.025L9.01 18.54l-1.157-2.352-2.302-1.16 1.051-1.055h1.871l2.67-2.637L4.5 8.086l1.366-1.391L13.94 8.53l3.826-3.776c.417-.359 1.039-.335 1.427.055s.412 1.013.055 1.432z" fill="currentColor" fill-rule="evenodd"/></svg>`

**Comment by fuzhenn** [comment]
after merging #21 , added [a test case](https://github.com/fuzhenn/node-svg2img/blob/master/test/specs.js#L113) and published as v0.6.1.
Thanks @RGunning !
